src/prepared-page.ts

```diff
@@ -0,0 +1,29 @@
+import { Page } from 'playwright';
+import { PreparedPage } from './types';
+
+/**
+ * Modify playwright or puppeteer page object to add additional properties
+ *
+ * @param page page object
+ */
+export function getPreparedPage( page ): PreparedPage< Page > {
+	// Initialize a flag to track whether the event listener is attached
+	if ( ! page._statusCodeListenerAttached ) {
+		page._statusCodeListenerAttached = true;
+
+		page.on( 'response', async response => {
+			if ( response.url() === page._gotoUrl ) {
+				page._statusCode = response.status();
+			}
+		} );
+	}
+
+	page.gotoWithStatus = async function ( url, options ) {
+		this._statusCode = null;
+		this._gotoUrl = url;
+
+		return await page.goto.call( this, url, options );
+	};
+
+	return page;
+}
```

Comment:
This only works because `page` is type `any`. I'd be more comfortable adding the method we need from `Page` to the `PageInterface`, so that this can expect a more explicit type.

Even better if we can just import the `Page` type from playwright without importing the whole lib. :)